exchanges/stream/websocket.go

```diff
@@ -127,47 +133,52 @@ func (w *Websocket) Setup(s *WebsocketSetup) error {
 	}
 	w.setEnabled(s.ExchangeConfig.Features.Enabled.Websocket)
 
-	if s.Connector == nil {
-		return fmt.Errorf("%s %w", w.exchangeName, errWebsocketConnectorUnset)
-	}
-	w.connector = s.Connector
-
-	if s.Subscriber == nil {
-		return fmt.Errorf("%s %w", w.exchangeName, errWebsocketSubscriberUnset)
-	}
-	w.Subscriber = s.Subscriber
+	w.useMultiConnectionManagement = s.UseMultiConnectionManagement
 
-	if w.features.Unsubscribe && s.Unsubscriber == nil {
-		return fmt.Errorf("%s %w", w.exchangeName, errWebsocketUnsubscriberUnset)
-	}
-	w.connectionMonitorDelay = s.ExchangeConfig.ConnectionMonitorDelay
-	if w.connectionMonitorDelay <= 0 {
-		w.connectionMonitorDelay = config.DefaultConnectionMonitorDelay
-	}
-	w.Unsubscriber = s.Unsubscriber
-
-	if s.GenerateSubscriptions == nil {
-		return fmt.Errorf("%s %w", w.exchangeName, errWebsocketSubscriptionsGeneratorUnset)
-	}
-	w.GenerateSubs = s.GenerateSubscriptions
+	if !w.useMultiConnectionManagement {
```

Comment:
Are we saying that we can't make all websockets use a single connection inside connection manager straight away, and remove the concept of "multi or not" straight away?

I can't see the connections being that invasive yet.

Reply:
Maybe I am not understanding the correct implementation of what you are trying to say. If you have time send me a diff here on the code block or open a PR against my branch and I will get to it monday. If it's a simple change then fo shiz.